### ISSUE TITLE
Wrong inputs removed on using remove from useFieldArray

### ISSUE THREAD

**Issue by danedavid** [react-hook-form/react-hook-form#3720]
**TLDR;** From a list of inputs, I remove the first input, but the input from the end is getting removed. Not sure if I'm not doing something as intended, but I've followed all rules specified here: https://react-hook-form.com/api/#useFieldArray

**To Reproduce**

1. Go to https://codesandbox.io/s/usefieldarray-wfbjt
2. Click on the 'x' button corresponding to input "balloon"
3. The second row, with value "goose" gets removed

**Expected behavior**
The input having "balloon" should be removed

**PS:** Great work on the library! Been using for some time and absolutely love the refreshing approach!

**Comment by bluebill1049** [comment]
hey @danedavid looks like you are missing `defaultValue` 👍

**Comment by danedavid** [comment]
@bluebill1049 It behaves the same way with `defaulValue` prop present. I've updated the CSB.

**Comment by bluebill1049** [comment]
<img width="585" alt="Screen Shot 2020-12-18 at 8 21 04 am" src="https://user-images.githubusercontent.com/10513364/102545150-fa03db80-4109-11eb-9621-d46a6e493cf3.png">

look closer this is not correct usage. check out the example folder for array of array.

**Comment by danedavid** [comment]
@bluebill1049 Are you talking about this: https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-x7btr?file=/src/fieldArray.js
But that looks like an example of nested `useFieldArray`, i.e., there are 2 sets of `useFieldArray`s. My example just has an array, each of which has arrays of input fields. The outer array is not modifiable and does not have fields in them.

Could you kindly let me know what is not a correct usage in the above screenshot? I assumed it would be mapping from `defaultValues` instead of `fields`, but like I said, there is no `fields` at that level. The `defaultValues.boards` is a set of constant, unchanging data.

**Comment by bluebill1049** [comment]
hey @danedavid you should convert the following map into a useFieldArray as well instead of mapping the default values straight way. 

```
       // defaultValues.boards this should be a field array.
       {defaultValues.boards.map((board, boardIndex) => {
          return (
            <Nominees
              key={board.id}
              board={board}
              boardIndex={boardIndex}
              control={control}
              register={register}
            />
          );
        })}
 ````
